scripts/package/postinstall.sh

```diff
@@ -99,10 +99,33 @@ finish_permissions() {
   # This prevents the service (running as root) from assigning ownership to
   # the root user. By doing so, we allow the user to switch to observiq-otel-collector
   # user for 'non root' installs.
-  touch /opt/observiq-otel-collector/log/collector.log
-  chown observiq-otel-collector:observiq-otel-collector /opt/observiq-otel-collector/log/collector.log
+  mkdir -p /opt/observiq-otel-collector/supervisor_storage
+  touch /opt/observiq-otel-collector/supervisor_storage/agent.log
+  touch /opt/observiq-otel-collector/supervisor_storage/effective.yaml
+  chown observiq-otel-collector:observiq-otel-collector /opt/observiq-otel-collector/supervisor_storage/agent.log
+  chown observiq-otel-collector:observiq-otel-collector /opt/observiq-otel-collector/supervisor_storage/effective.yaml
```

Comment:
I think the supervisor_storage dir, agent log, and effective config could be defined by Goreleaser.

```yaml
      - dst: /opt/observiq-otel-collector/supervisor_storage
        type: dir
        file_info:
          mode: 0750
          owner: observiq-otel-collector
          group: observiq-otel-collector
      - dst: /opt/observiq-otel-collector/supervisor_storage/effective.yaml
        type: ghost
      - dst: /opt/observiq-otel-collector/supervisor_storage/agent.log
        type: ghost
```

For the ghost files, we should ensure their permissions end up being 0640 or 0600 after they are created by the agent. This is not something the scripts or Goreleaser should handle.